src/edit.rs

```diff
@@ -48,19 +49,32 @@ pub fn start(cli: &ArgMatches<'_>) {
         | if cli.is_present("tmp") { Flags::TMP } else { Flags::NULL };
 
     // NOTE: unwrap can't faile here, because PROFILE_NAME is required
-    let profile_name = get_name1(cli.value_of("PROFILE_NAME").unwrap());
+    let profile_name = cli.value_of("PROFILE_NAME").unwrap();
+
     debug!("profile name: {}", profile_name);
 
-    let system_profile = profile_path!(SYSTEM / &profile_name);
-    debug!("system profile: {}", system_profile.to_string_lossy());
+    let user_profile = Profile::new(
+        profile_name,
+        ProfileFlags::LOOKUP_USER | ProfileFlags::DENY_BY_PATH | ProfileFlags::ASSUME_EXISTENCE,
+    )
+    .unwrap();
 
-    let user_profile = profile_path!(USER / &profile_name);
-    debug!("user profile: {}", user_profile.to_string_lossy());
+    debug!("user profile: {}", user_profile.full_name());
 
-    if flags.contains(Flags::TMP) {
-        prepare_tmp_edit(&user_profile, &system_profile, flags);
-    } else {
-        prepare_edit(&user_profile, &system_profile, flags);
+    let system_profile = Profile::new(
+        profile_name,
+        ProfileFlags::LOOKUP_SYSTEM | ProfileFlags::DENY_BY_PATH | ProfileFlags::ASSUME_EXISTENCE,
+    )
+    .unwrap();
+
+    debug!("system profile: {}", &system_profile.full_name());
+
+    if let (Some(user_path), Some(system_path)) = (&user_profile.path(), &system_profile.path()) {
```

Comment:
What are the reasons for borrowing here (line 70 and 72)?

Reply:
For userprofile and system, it is a convention I use in fast changing code to borrow values instead of owning. This comes in handy sometimes. But I have updated the code to your suggestions.